crates/driver/src/infra/api/error.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    crate::{
+        domain::competition::{self, quote, solution},
+        infra::api,
+    },
+    hyper::StatusCode,
+    serde::Serialize,
+};
+
+#[derive(Debug, Clone, Copy, Serialize)]
+#[serde(rename_all = "PascalCase")]
+enum Type {
+    QuotingFailed,
+    SolverFailed,
+    SolutionNotFound,
+    DeadlineExceeded,
+    SimulationFailed,
+    Unknown,
+    TransactionPublishingFailed,
+    InvalidAuctionId,
+    MissingSurplusFee,
+    LiquidityError,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Error {
+    error_type: Type,
+    description: &'static str,
+}
+
+impl From<Type> for (StatusCode, axum::Json<Error>) {
+    fn from(value: Type) -> Self {
+        let (description, status_code) = match value {
+            Type::QuotingFailed => ("No valid quote found", StatusCode::BAD_REQUEST),
```

Comment:
To me, this is more that the server couldn't find a quote, not that the request made by the client was bad, I think a 404 is more appropriate:
```suggestion
            Type::QuotingFailed => ("No valid quote found", StatusCode::NOT_FOUND),
```